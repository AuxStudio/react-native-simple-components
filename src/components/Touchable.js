import React from "react";
import {
    View,
    TouchableNativeFeedback,
    TouchableWithoutFeedback,
    TouchableOpacity,
    Platform,
} from "react-native";

export default (Touchable = props => {
    /* 
        PROPTYPES

        androidRipple: PropTypes.bool,  // flag to display android ripple
        androidRippleColor: PropTypes.string,
        androidRippleBorderless: PropTypes.bool, // android ripple will extend beyond object boundaries
        onPress: PropTypes.func.isRequired,
        disableFeedback: PropTypes.bool, // will render TouchableWithoutFeedback
        children: PropTypes.node,
        // style: PropTypes.node,

    */

    const touchable =
        props.androidRipple && Platform.OS === "android" ? (
            <TouchableNativeFeedback
                onPress={props.onPress}
                background={TouchableNativeFeedback.Ripple(
                    props.androidRippleColor,
                    props.androidRippleBorderless
                )}>
                <View style={props.style}>{props.children}</View>
            </TouchableNativeFeedback>
        ) : props.disableFeedback ? (
            <TouchableWithoutFeedback onPress={props.onPress}>
                <View style={props.style}>{props.children}</View>
            </TouchableWithoutFeedback>
        ) : (
            <TouchableOpacity onPress={props.onPress} style={props.style}>
                {props.children}
            </TouchableOpacity>
        );

    return touchable;
});
