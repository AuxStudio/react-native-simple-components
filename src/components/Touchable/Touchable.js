import React from 'react';
import {
  View,
  TouchableNativeFeedback,
  TouchableWithoutFeedback,
  TouchableOpacity,
  Platform,
} from 'react-native';

export default (Touchable = (props) => {
  /* 
        PROPTYPES

        androidRipple: PropTypes.bool,  // flag to display android ripple
        androidRippleColor: PropTypes.string,
        androidRippleBorderless: PropTypes.bool, // android ripple will extend beyond object boundaries
        onPress: PropTypes.func.isRequired,
        onLongPress: PropTypes.func,
        disableFeedback: PropTypes.bool, // will render TouchableWithoutFeedback
        children: PropTypes.node,
        // style: PropTypes.node,

    */

  const touchable =
    props.androidRipple && Platform.OS === 'android' ? (
      <TouchableNativeFeedback
        onPress={props.onPress}
        onLongPress={props.onLongPress}
        background={TouchableNativeFeedback.Ripple(
          props.androidRippleColor,
          props.androidRippleBorderless,
        )}
      >
        <View style={props.style}>{props.children}</View>
      </TouchableNativeFeedback>
    ) : props.disableFeedback ? (
      <TouchableWithoutFeedback onPress={props.onPress} onLongPress={props.onLongPress}>
        <View style={props.style}>{props.children}</View>
      </TouchableWithoutFeedback>
    ) : (
      <TouchableOpacity onPress={props.onPress} onLongPress={props.onLongPress} style={props.style}>
        {props.children}
      </TouchableOpacity>
    );

  return touchable;
});
