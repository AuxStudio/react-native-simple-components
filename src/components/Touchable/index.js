import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  TouchableNativeFeedback,
  TouchableWithoutFeedback,
  TouchableOpacity,
  Platform,
  ViewPropTypes,
} from 'react-native';

const propTypes = {
  androidRipple: PropTypes.bool, // flag to display android ripple effect
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool, // android ripple will extend beyond object boundaries
  onPress: PropTypes.func,
  onLongPress: PropTypes.func,
  disableFeedback: PropTypes.bool, // will render TouchableWithoutFeedback
  disabled: PropTypes.bool,
  children: PropTypes.node,
  style: ViewPropTypes.style,
  testID: PropTypes.string,
};

const defaultProps = {};

const Touchable = ({
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  onPress,
  onLongPress,
  disableFeedback,
  disabled,
  children,
  style,
  testID,
}) => {
  let touchableComponent;

  if (androidRipple && Platform.OS === 'android') {
    touchableComponent = (
      <TouchableNativeFeedback
        onPress={onPress}
        onLongPress={onLongPress}
        background={TouchableNativeFeedback.Ripple(androidRippleColor, androidRippleBorderless)}
        disabled={disabled}
        testID={testID}
      >
        <View style={style}>{children}</View>
      </TouchableNativeFeedback>
    );
  } else if (disableFeedback) {
    touchableComponent = (
      <TouchableWithoutFeedback
        onPress={onPress}
        onLongPress={onLongPress}
        disabled={disabled}
        testID={testID}
      >
        <View style={style}>{children}</View>
      </TouchableWithoutFeedback>
    );
  } else {
    touchableComponent = (
      <TouchableOpacity
        onPress={onPress}
        onLongPress={onLongPress}
        style={style}
        disabled={disabled}
        testID={testID}
      >
        {children}
      </TouchableOpacity>
    );
  }

  return touchableComponent;
};

Touchable.propTypes = propTypes;
Touchable.defaultProps = defaultProps;

export default Touchable;
