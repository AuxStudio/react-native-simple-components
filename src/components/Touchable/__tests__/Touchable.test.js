/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Touchable from '../Touchable';

it('renders a Touchable', () => {
  expect(
    renderer.create(
      <Touchable
        onPress={jest.fn()}
        onLongPress={jest.fn()}
        androidRipple
        androidRippleColor="red"
        androidRippleBorderless
        style={{ backgroundColor: 'red' }}
      >
        <View />
      </Touchable>,
    ),
  ).toMatchSnapshot();
});

it('renders a Touchable with no feedback', () => {
  expect(
    renderer.create(
      <Touchable disabledFeedback>
        <View />
      </Touchable>,
    ),
  ).toMatchSnapshot();
});

it('renders a disabled Touchable', () => {
  expect(
    renderer.create(
      <Touchable disabled>
        <View />
      </Touchable>,
    ),
  ).toMatchSnapshot();
});

it('renders a Touchable with no props', () => {
  expect(renderer.create(<Touchable />)).toMatchSnapshot();
});
/* eslint-enable */
