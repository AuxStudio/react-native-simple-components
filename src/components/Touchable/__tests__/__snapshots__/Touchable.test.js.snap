// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable renders with all props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "opacity": 1,
    }
  }
>
  <View />
</View>
`;

exports[`Touchable renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;

exports[`Touchable renders with no feedback 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View />
</View>
`;
